src/BuildScriptGeneratorCli/Commands/BuildpackDetectCommand.cs

```diff
@@ -4,35 +4,77 @@
 // --------------------------------------------------------------------------------------------
 
 using System;
+using System.CommandLine;
+using System.CommandLine.IO;
 using System.IO;
 using System.Linq;
-using McMaster.Extensions.CommandLineUtils;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Microsoft.Oryx.BuildScriptGenerator;
 using Microsoft.Oryx.BuildScriptGenerator.Common;
+using Microsoft.Oryx.BuildScriptGeneratorCli.Commands;
 
 namespace Microsoft.Oryx.BuildScriptGeneratorCli
 {
-    [Command(Name, Description = "Determine whether Oryx can be applied as a buildpack to an app in the current " +
-        "working directory.")]
     internal class BuildpackDetectCommand : CommandBase
     {
-        public const string Name = "buildpack-detect";
-
         // CodeDetectFail @ https://github.com/buildpack/lifecycle/blob/master/detector.go
         public const int DetectorFailCode = 100;
 
-        [Argument(0, Description = "The source directory.")]
+        public BuildpackDetectCommand()
+        {
+        }
+
+        public BuildpackDetectCommand(BuildpackDetectCommandProperty input)
+        {
+            this.PlatformDir = input.PlatformDir;
+            this.SourceDir = input.SourceDir;
+            this.PlanPath = input.PlanPath;
+            this.LogFilePath = input.LogFilePath;
+            this.DebugMode = input.DebugMode;
+        }
+
         public string SourceDir { get; set; }
 
-        [Option("--platform-dir <dir>", CommandOptionType.SingleValue, Description = "Platform directory path.")]
         public string PlatformDir { get; set; }
 
-        [Option("--plan-path <path>", CommandOptionType.SingleValue, Description = "Build plan TOML path.")]
         public string PlanPath { get; set; }
 
+        public static Command Export()
+        {
+            var sourceDirArgument = new Argument<string>("SourceDir", "The source directory.");
+            var platformDirOption = new Option<string>("--platform-dir", "Platform directory path.");
```

Comment:
nit: we should add `--platform-dir` and `--plan-path` to the `OptionTemplates` file